core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -203,19 +204,28 @@ private void notifyAdvertStart(AdvertBreak advertBreak) {
 
     @Override
     public void onPositionDiscontinuity(int reason) {
-        if (reason != Player.DISCONTINUITY_REASON_AD_INSERTION || player == null || adPlaybackState == null) {
+        if (player == null || adPlaybackState == null || advertsDisabled) {
             // We need all of the above to be able to respond to advert events.
             return;
         }
 
-        if (isPlayingAdvert()) {
-            notifyAdvertEnd(advertBreaks.get(adGroupIndex));
-            adPlaybackState = adPlaybackState.withPlayedAd(adGroupIndex, adIndexInGroup);
+        if (reason == Player.DISCONTINUITY_REASON_SEEK) {
+            long contentPosition = player.getContentPosition();
+            adPlaybackState = AvailableAdverts.from(contentPosition, advertBreaks, this.adPlaybackState);
```

Comment:
Why do we need to do this here ? This seems potentially taxing

Reply:
We spoke about this offline. This is needed because when seeking backwards we need to be able to mark an advert as available to play again, that way when they seek forwards or the player naturally hits that time it will trigger the advert.